### PR TITLE
[Clang importer] Initial support for swift_newtype / swift_wrapper

### PR DESCRIPTION
**Pull request by milseman**
#### What's in this pull request?

Adds initial support for swift_newtype(struct), and swift_newtype(enum) by treating as swift_newtype(struct). 
#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by milseman** [comment]
@swift-ci please test

**Comment by milseman** [comment]
pexpect.TIMEOUT: Timeout exceeded.
Failing Tests (1):
    swift-package-tests :: repl/test-repl-glibc.py

This seems unrelated and a problem with the CI. @shahmishal what do you think? Should I kick it off again? Are you familiar with this test?

**Comment by shahmishal** [comment]
@milseman This test failure is related to https://bugs.swift.org/browse/SR-1109

**Comment by milseman** [comment]
@DougGregor, can you please review these changes?

File: lib/ClangImporter/ClangImporter.cpp [hunk 1]
```diff
@@ -2538,6 +2552,31 @@ auto ClangImporter::Implementation::importFullName(
...
+  // swift_newtype-ed declarations may have common words with the type name
+  // stripped.
+  if (auto newtypeDecl = findSwiftNewtype(D)) {
+    SmallString<32> newNameBuffer;
+    auto newtypeNameWords = camel_case::getWords(newtypeDecl->getName());
+    auto declNameWords = camel_case::getWords(baseName);
+
+    // Scan the decl name, and drop any words that appear in order in the type
+    // name
+    for (auto wI = declNameWords.begin(), skipWI = newtypeNameWords.begin();
```
Comment (DougGregor) [inline lib/ClangImporter/ClangImporter.cpp]: Is there truly no utility within the enum-handling code to compare common prefixes between two names?

**Comment by DougGregor** [comment]
Please add a test that uses the imported results, so we're certain that name lookup finds these implicitly-imported members.

File: lib/ClangImporter/ClangImporter.cpp (hunk 1 above)
Comment (milseman) [inline lib/ClangImporter/ClangImporter.cpp]: Not in the spirit of SE-0033, in which there's the example:
extern NSErrorDomain const NSCocoaErrorDomain ===> NSErrorDomain.cocoa

In this case, it's not prefix stripping but rather common-word stripping. I'm happy to make a utility in StringExtras or wherever that can help expose, share, or unify these tasks, but I wasn't able to find a direct analogue to what I was trying to do. 

In general, I am interested in more camel_case utilities that can also help unify with the import-as-member inference system (which has its own handful of helpers) and enum handling (which is just prefix).

Comment (milseman) [inline lib/ClangImporter/ClangImporter.cpp]: In the meantime, it's worth me extracting this into a static local function, to not overly pollute importFullName and give it a meaningful name.

Comment (milseman) [inline lib/ClangImporter/ClangImporter.cpp]: We discussed offline, and we're going to go against the proposal and only do prefix stripping. I'll add this to the list of things to review for the proposal amend.

**Comment by milseman** [comment]
@swift-ci please test

**Pull request merged**